### PR TITLE
FFWEB-2460: export preview (product view) RELEASE/3.x 

### PR DESCRIPTION
**Pull request by grzegorz-jamroz**
- Solves issue: 
    -  [Preview of the exported values](https://fact-finder.atlassian.net/browse/FFWEB-2460)
- Description: 
    -  added preview of the exported values
    - preview is avaliable in Product view
- Tested with Magento editions/versions: 
    - 2.4.4
- Tested with PHP versions: 
    - 7.4

### CODE REVIEW

File: src/Block/Adminhtml/Product/Edit/Button/ExportPreview.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Omikron\Factfinder\Block\Adminhtml\Product\Edit\Button;
+
+use Magento\Catalog\Block\Adminhtml\Product\Edit\Button\Generic;
+
+/**
+ * Class AddAttribute
```
Comment (a-laurowski) [inline src/Block/Adminhtml/Product/Edit/Button/ExportPreview.php]: imo such docblocks are completely useless

File: src/etc/di.xml [hunk 2]
```diff
@@ -30,6 +30,27 @@
             <argument name="instanceName" xsi:type="string">Magento\Catalog\Model\ResourceModel\Product\Attribute\Collection</argument>
         </arguments>
     </virtualType>
+    <virtualType name="Omikron\Factfinder\Model\Export\ExportPreviewProductFeed" type="Omikron\Factfinder\Model\Export\Feed">
```
Comment (a-laurowski) [inline src/etc/di.xml]: can we avoid defining the columns again? Can a virtual type extend already existing type?

File: src/Utilities/Validator/ExportPreviewValidator.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    /** @var ProductRepositoryInterface */
+    private $productRepository;
+
+    /** @var ConfigurableType */
+    private $configurableType;
+
+    public function __construct(
+        int $entityId,
+        ProductRepositoryInterface $productRepository,
```
Comment (a-laurowski) [inline src/Utilities/Validator/ExportPreviewValidator.php:25]: i don't know if there i a rule for that but i personally sort the arguments that are primitives, to be the last in the whole argument list

File: src/Model/Stream/Json.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+declare(strict_types=1);
+
+namespace Omikron\Factfinder\Model\Stream;
+
+use Omikron\Factfinder\Api\StreamInterface;
+
+class Json implements StreamInterface
+{
+    /** @var array */
```
Comment (a-laurowski) [inline src/Model/Stream/Json.php]: redundant comment

File: src/Model/Export/Catalog/ExportPreviewProducts.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+class ExportPreviewProducts implements \IteratorAggregate
+{
+    /** @var ProductRepositoryInterface */
+    private $productRepository;
+
+    /** @var RequestInterface */
+    private $request;
+
+    /** @var int */
```
Comment (a-laurowski) [inline src/Model/Export/Catalog/ExportPreviewProducts.php]: redundant comment

**Review by a-laurowski — CHANGES_REQUESTED** [review]

File: src/Block/Adminhtml/Product/Edit/Button/ExportPreview.php (hunk 1 above)
Comment (grzegorz-jamroz) [reply src/Block/Adminhtml/Product/Edit/Button/ExportPreview.php]: removed 👍

File: src/Utilities/Validator/ExportPreviewValidator.php (hunk 3 above)
Comment (grzegorz-jamroz) [reply src/Utilities/Validator/ExportPreviewValidator.php:25]: I pushed `$entityId` as last argument👍

File: src/Model/Stream/Json.php (hunk 4 above)
Comment (grzegorz-jamroz) [reply src/Model/Stream/Json.php]: removed 👍

File: src/Model/Export/Catalog/ExportPreviewProducts.php (hunk 5 above)
Comment (grzegorz-jamroz) [reply src/Model/Export/Catalog/ExportPreviewProducts.php]: removed 👍

File: src/etc/di.xml (hunk 2 above)
Comment (grzegorz-jamroz) [reply src/etc/di.xml]: `Omikron\Factfinder\Model\Export\ExportPreviewProductFeed` now extends `Omikron\Factfinder\Model\Export\CatalogFeed` 👍

**Review by a-laurowski — APPROVED** [review]

**Pull request merged**